### ISSUE TITLE
Timeout when not running on GCE

### ISSUE THREAD

**Issue by CapacitorSet** [GoogleCloudPlatform/microservices-demo#685]
### Describe the bug 
When the application is deployed on [Kind](https://kind.sigs.k8s.io/) (and, I believe, any non-GCE platform), requests time out.

I think the root cause is that the frontend calls `getDeploymentDetails` on every request, which uses `cloud.google.com/go/compute/metadata` and makes a request to `http://169.254.169.254/computeMetadata/...`. This IP however does not exist on other platforms, causing delays of several minutes in HTTP requests.

This appears to be a regression from 09956ca4.

### To Reproduce 
1. Install Kind
2. Create a Kind node: `kind create cluster --kubeconfig=/tmp/config --name=cluster1`
3. In the Kind node, apply `release/kubernetes-manifests.yaml` from v0.3.5
4. Make a request to the frontend
5. Wait for the request to terminate; if no timeout is applied, it will return after 10 minutes with the homepage.

### Logs  
Frontend logs:
```
{
  "http.req.id": "8fccb933-80d9-4482-badf-a3963f35a13f",
  "http.req.method": "GET",
  "http.req.path": "/",
  "message": "Failed to fetch the name of the cluster in which the pod is runningGet \"http://169.254.169.254/computeMetadata/v1/instance/attributes/cluster-name\": dial tcp 169.254.169.254:80: i/o timeout",
  "session": "0706fb66-c1aa-4215-a93a-8db82f586e7c",
  "severity": "error",
  "timestamp": "2022-01-11T14:44:21.017021115Z"
}
{
  "http.req.id": "8fccb933-80d9-4482-badf-a3963f35a13f",
  "http.req.method": "GET",
  "http.req.path": "/",
  "message": "Failed to fetch the Zone of the node where the pod is scheduledGet \"http://169.254.169.254/computeMetadata/v1/instance/zone\": dial tcp 169.254.169.254:80: i/o timeout",
  "session": "0706fb66-c1aa-4215-a93a-8db82f586e7c",
  "severity": "error",
  "timestamp": "2022-01-11T14:47:22.011728459Z"
}
```

### Environment 
 - OS: Arch Linux
 - Kubernetes distribution, version: Kind v0.11.1

### Exposure 
This bug occurs on every deployment. I believe it to be the reason for end-to-end tests in our CI failing due to timeouts.

**Comment by saamalik** [comment]
I can confirm this behavior. I reverted to release `v0.3.1` to get around this. Thx!

**Comment by NimJay** [comment]
@CapacitorSet Thanks so much for reporting this issue — and for doing some root cause analysis. :) That helped a lot.

We just released `v0.3.6` of `microservices-demo` (Online Boutique) which should work fine on Kind (and other non-GCP clusters).
Sorry it took some time!
